### ISSUE TITLE
brotli_nginx fails to find nginx source

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/ofn-install#516]
> TASK [brotli_nginx : download and extract nginx source] ***********************************************************************************************************************************************************
> Wednesday 11 September 2019  08:39:05 +1000 (0:00:06.848)       0:17:27.175 *** 
> fatal: [staging.openfoodnetwork.org.au]: FAILED! => {"changed": false, "msg": "Failure downloading http://nginx.org/download/nginx-1.10.3 (Ubuntu).tar.gz, HTTP Error 404: Not Found"}

That file isn't there. I would also strongly recommend to download via https. I found this link:
https://nginx.org/download/nginx-1.10.3.tar.gz

**Comment by mkllnk** [comment]
Somehow the version output of nginx is different on this server.

```sh
ssh ofn-au-prod
nginx -v
# nginx version: nginx/1.16.1

ssh ofn-au-staging
nginx -v
# nginx version: nginx/1.10.3 (Ubuntu)
```

I think the role assumes that nginx has been installed from a custom ppa before.

**Comment by mkllnk** [comment]
I solved the version problem and got a new one:

> fatal: [staging.openfoodnetwork.org.au]: FAILED! => {"changed": true, "cmd": "./configure --with-cc-opt='-g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2' --with-ld-opt='-Wl,-Bsymbolic-functions -fPIE -pie -Wl,-z,relro -Wl,-z,now' --prefix=/usr/share/nginx --conf-path=/etc/nginx/nginx.conf --http-log-path=/var/log/nginx/access.log --error-log-path=/var/log/nginx/error.log --lock-path=/var/lock/nginx.lock --pid-path=/run/nginx.pid --http-client-body-temp-path=/var/lib/nginx/body --http-fastcgi-temp-path=/var/lib/nginx/fastcgi --http-proxy-temp-path=/var/lib/nginx/proxy --http-scgi-temp-path=/var/lib/nginx/scgi --http-uwsgi-temp-path=/var/lib/nginx/uwsgi --with-debug --with-pcre-jit --with-ipv6 --with-http_ssl_module --with-http_stub_status_module --with-http_realip_module --with-http_auth_request_module --with-http_addition_module --with-http_dav_module --with-http_geoip_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_image_filter_module --with-http_v2_module --with-http_sub_module --with-http_xslt_module --with-stream --with-stream_ssl_module --with-mail --with-mail_ssl_module --with-threads --with-compat --add-dynamic-module=/usr/local/src/ngx_brotli", "delta": "0:00:00.080414", "end": "2019-09-10 23:47:02.432122", "msg": "non-zero return code", "rc": 1, "start": "2019-09-10 23:47:02.351708", "stderr": "", "stderr_lines": [], "stdout": "./configure: error: invalid option \"--with-compat\"", "stdout_lines": ["./configure: error: invalid option \"--with-compat\""]}

**Comment by mkllnk** [comment]
Okay, I looked into upgrading nginx but it's the latest Ubuntu 16 version already. I'm blocked on provisioning au-staging and don't have time to solve it right now. Putting it into dev-ready.

**Comment by sauloperez** [comment]
Have you considered creating it from scratch or close to that? we know running it on a blank server ofn-install sets the desired state.

**Comment by mkllnk** [comment]
You mean provisioning a completely new staging server for au? That's too much hassle right now because Kirsten wants to keep the current data. But we should do it at some point. Another opportunity for Ubuntu18?

**Comment by mkllnk** [comment]
It appears that the same is now happening to people in New Zealand.

**Comment by openfoodnetworknz** [comment]
Yup, this error when running site.yml on a blank server:

```
TASK [brotli_nginx : download and extract nginx source] **********************************************************************************************************
Monday 16 September 2019  04:14:51 +0000 (0:00:00.191)       0:09:30.445 ******
fatal: [openfoodnetwork.org.nz]: FAILED! => {"changed": false, "msg": "Failure downloading http://nginx.org/download/nginx-1.10.3 (Ubuntu).tar.gz, An unknown error occurred: URL can't contain control characters. '/download/nginx-1.10.3 (Ubuntu).tar.gz' (found at least ' ')"}
```

I can get round it by installing nginx manually using apt-get to install and then running ansible again.

**Comment by RachL** [comment]
> You mean provisioning a completely new staging server for au? That's too much hassle right now because Kirsten wants to keep the current data. But we should do it at some point. Another opportunity for Ubuntu18?

@mkllnk is it because AU staging has German data? We can re-use product import to get them back. ping @kirstenalarsen

**Comment by Matt-Yorkley** [comment]
Is this an apt update issue? It was tested with the current Ubuntu 16 version of nginx... And it looks like manually installing nginx via apt fixes it? :thinking:

**Comment by Matt-Yorkley** [comment]
l take a look as soon as I get some time.

**Comment by kirstenalarsen** [comment]
![image](https://user-images.githubusercontent.com/2911176/65030871-a5f8cd00-d983-11e9-93d5-75c1798b4f81.png)
![image](https://user-images.githubusercontent.com/2911176/65030901-b14bf880-d983-11e9-9371-868c6afb0340.png)

**Comment by Matt-Yorkley** [comment]
I'll look at this today. There's a few things: 

- The download should be using https. Easy to fix.
- I don't think any of our Ubuntu 16 servers should be installing nginx `1.10.3` via apt. I'll investigate that.
- It looks like the output  of: `nginx -v` with version `1.10.3` adds the string ` (Ubuntu)` at the end, which we register at some point. I think it just needs the string to be stripped to `1.10.3` and it'll be fixed.

**Comment by Matt-Yorkley** [comment]
there's definitely some weird apt stuff going on...

Aus Staging:
![nginx1](https://user-images.githubusercontent.com/9029026/65041222-28ff3080-d94e-11e9-81b4-2e93b620a2d9.png)
UK Prod:
![nginx2](https://user-images.githubusercontent.com/9029026/65041223-28ff3080-d94e-11e9-8a66-1b94f348d75d.png)

**Comment by Matt-Yorkley** [comment]
Ah! it's already been done in https://github.com/openfoodfoundation/ofn-install/pull/517/files :muscle:

**Comment by Matt-Yorkley** [comment]
There are some weird `apt` conflicts on Aus staging, and the package is way behind what it should be:

![nginx3](https://user-images.githubusercontent.com/9029026/65041774-5bf5f400-d94f-11e9-905f-175194f7ecc8.png)

**Comment by Matt-Yorkley** [comment]
OK, Aus Staging is now fixed, and the other issues should be fixed by @mkllnk 's work in #517

**Comment by openfoodnetworknz** [comment]
@Matt-Yorkley Well, it got past that task, but now I'm getting a new error (even after testing with a fresh ubuntu build):

```TASK [brotli_nginx : configure build] ****************************************************************************************************************************************************************************************************************************
Wednesday 18 September 2019  05:09:04 +0000 (0:00:00.054)       0:01:49.298 ***
fatal: [openfoodnetwork.org.nz]: FAILED! => {"changed": true, "cmd": "./configure --with-cc-opt='-g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2' --with-ld-opt='-Wl,-Bsymbolic-functions -fPIE -pie -Wl,-z,relro -Wl,-z,now' --prefix=/usr/share/nginx --conf-path=/etc/nginx/nginx.conf --http-log-path=/var/log/nginx/access.log --error-log-path=/var/log/nginx/error.log --lock-path=/var/lock/nginx.lock --pid-path=/run/nginx.pid --http-client-body-temp-path=/var/lib/nginx/body --http-fastcgi-temp-path=/var/lib/nginx/fastcgi --http-proxy-temp-path=/var/lib/nginx/proxy --http-scgi-temp-path=/var/lib/nginx/scgi --http-uwsgi-temp-path=/var/lib/nginx/uwsgi --with-debug --with-pcre-jit --with-ipv6 --with-http_ssl_module --with-http_stub_status_module --with-http_realip_module --with-http_auth_request_module --with-http_addition_module --with-http_dav_module --with-http_geoip_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_image_filter_module --with-http_v2_module --with-http_sub_module --with-http_xslt_module --with-stream --with-stream_ssl_module --with-mail --with-mail_ssl_module --with-threads --with-compat --add-dynamic-module=/usr/local/src/ngx_brotli", "delta": "0:00:00.074696", "end": "2019-09-18 17:09:04.487504", "msg": "non-zero return code", "rc": 1, "start": "2019-09-18 17:09:04.412808", "stderr": "", "stderr_lines": [], "stdout": "./configure: error: invalid option \"--with-compat\"", "stdout_lines": ["./configure: error: invalid option \"--with-compat\""]}```

Ideas?

**Comment by Matt-Yorkley** [comment]
Yes. It should be `1.16.1` if you're on Ubuntu 16 and have updated `apt`.

**Comment by romale** [comment]
I've this issue too on ubuntu 16
There're error description
https://trac.nginx.org/nginx/ticket/1377
"The --with-compat configure option is available in nginx 1.11.5 and later."
